ui/guest/src/react/I18nProvider.tsx

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright (C) 2007-2025 Crafter Software Corporation. All Rights Reserved.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as published by
+ * the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React, { PropsWithChildren, useState } from 'react';
+import { IntlShape, RawIntlProvider } from 'react-intl';
+import { getCurrentIntl } from '../utils/i18n';
+
+export function I18nProvider(props: PropsWithChildren<{}>) {
+  const [intl] = useState<IntlShape>(getCurrentIntl());
```

Comment:
At the moment this is slightly pointless, but I guess serves as foundation for later.
```suggestion
  const [intl] = useState<IntlShape>(() => getCurrentIntl());
```